skimage/exposure/tests/test_exposure.py

```diff
@@ -362,16 +362,56 @@ def test_adapthist_alpha():
     assert_almost_equal(norm_brightness_err(full_scale, adapted), 0.0248, 3)
 
 
+def test_adapthist_grayscale_Nd():
+    """
+    Test for n-dimensional consistency with float images
+    Note: Currently if img.ndim == 3, img.shape[2] > 4 must hold for the image
+    not to be interpreted as a color image by @adapt_rgb
+    """
+    # take 2d image, subsample and stack it
+    img = util.img_as_float(data.astronaut())
+    img = rgb2gray(img)
+    a = 15
+    img2d = util.img_as_float(img[0:-1:a, 0:-1:a])
+    img3d = np.array([img2d] * (img.shape[0] // a))
+
+    # apply CLAHE
+    adapted2d = exposure.equalize_adapthist(img2d,
+                                            kernel_size=5,
+                                            clip_limit=0.05)
+    adapted3d = exposure.equalize_adapthist(img3d,
+                                            kernel_size=5,
+                                            clip_limit=0.05)
+
+    # check that dimensions of input and output match
+    assert img2d.shape == adapted2d.shape
+    assert img3d.shape == adapted3d.shape
+
+    # check that the result from the stack of 2d images is similar
+    # to the underlying 2d image
+    assert np.mean(np.abs(adapted2d
+                          - adapted3d[adapted3d.shape[0] // 2])) < 0.02
+
+
 def test_adapthist_borders():
     """Test border processing
     """
     img = rgb2gray(util.img_as_float(data.astronaut()))
-    adapted = exposure.equalize_adapthist(img, 11)
-    width = 42
-    # Check last columns are procesed
-    assert norm_brightness_err(adapted[:, -width], img[:, -width]) > 1e-3
-    # Check last rows are procesed
-    assert norm_brightness_err(adapted[-width, :], img[-width, :]) > 1e-3
+
+    # maximize difference between orig and processed img
+    img /= 100.
+    img[img.shape[0] // 2, img.shape[1] // 2] = 1.
+
+    # check borders are processed for different kernel sizes
+    border_index = -1
+    for kernel_size in range(51, 71, 2):
+        adapted = exposure.equalize_adapthist(img, kernel_size, clip_limit=0.5)
+        # Check last columns are processed
+        assert norm_brightness_err(adapted[:, border_index],
```

Comment:
I don't understand these tests here. Shouldn't we compare the same columns on the two images? I wonder if there was a bug here (not introduced by your PR, but let's fix it if it is indeed a bug).

Reply:
You're completely right! I think when I added the minus to the border_index value I forgot to take it away from the index into the original image. Good catch!

Reply:
There was a problem in this test before, however it was not the comparison of different columns (which I had introduced). Namely that column -42 was compared (together with a slightly misleading `width` variable) and that hadn't catched the remaining border artefact in some of the last columns.